homeassistant/components/notion/__init__.py

```diff
@@ -0,0 +1,272 @@
+        """Refresh Notion sensor data."""
+        _LOGGER.debug('Refreshing Notion sensor data')
+        await notion.async_update()
+        async_dispatcher_send(hass, TOPIC_DATA_UPDATE)
+
+    hass.data[DOMAIN][DATA_LISTENER][
+        config_entry.entry_id] = async_track_time_interval(
+            hass,
+            refresh,
+            timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL]))
+
+    return True
+
+
+async def async_unload_entry(hass, config_entry):
+    """Unload a Notion config entry."""
+    hass.data[DOMAIN][DATA_CLIENT].pop(config_entry.entry_id)
+    cancel = hass.data[DOMAIN][DATA_LISTENER].pop(config_entry.entry_id)
+    cancel()
+
+    for component in ('binary_sensor', 'sensor'):
+        await hass.config_entries.async_forward_entry_unload(
+            config_entry, component)
+
+    return True
+
+
+class Notion:
+    """Define a class to handle the Notion API."""
+
+    def __init__(self, client):
+        """Initialize."""
+        self._client = client
+        self.bridges = []
+        self.sensors = []
+        self.systems = []
+        self.tasks = []
+
+    async def async_update(self, *, initialize=False):
+        """Get the latest Notion data."""
+        from aionotion.errors import NotionError
+
+        tasks = {
+            'bridges': self._client.bridge.async_all(),
+            'sensors': self._client.sensor.async_all(),
+            'tasks': self._client.task.async_all(),
+        }
+        if initialize:
+            tasks['systems'] = self._client.system.async_all()
+
+        results = await asyncio.gather(*tasks.values(), return_exceptions=True)
+        for attr, result in zip(tasks, results):
+            if isinstance(result, NotionError):
+                _LOGGER.error(
+                    'There was an error while updating %s: %s', attr, result)
+                continue
+
+            setattr(self, attr, result)
+
+
+class NotionEntity(Entity):
+    """Define a base Notion entity."""
+
+    def __init__(
+            self,
+            notion,
+            task,
+            sensor,
+            bridge,
+            system,
+            name,
+            device_class):
+        """Initialize the entity."""
+        self._async_unsub_dispatcher_connect = None
+        self._bridge = bridge
+        self._device_class = device_class
+        self._name = name
+        self._notion = notion
+        self._sensor = sensor
+        self._state = None
+        self._system = system
+        self._task = task
+
+        self._attrs = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_BRIDGE_MODE: self._bridge['mode'],
+            ATTR_BRIDGE_NAME: self._bridge['name'],
+            ATTR_SYSTEM_MODE: self._system['mode'],
+            ATTR_SYSTEM_NAME: self._system['name'],
+        }
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._task['id'] in [t['id'] for t in self._notion.tasks]
```

Comment:
You could make this slightly faster by using `any(self._task['id'] == t['id'] for t in self._notion.tasks)` as it will bail out as soon as it's found. By testing it like you're doing now, you're going to first build the whole list instead of stopping as soon as it's found.